mongodb-plugin/src/main/java/io/cdap/plugin/batch/source/BSONObjectToRecordTransformer.java

```diff
@@ -23,45 +23,29 @@
 import org.bson.BSONObject;
 import org.bson.types.BasicBSONList;
 
-import java.io.IOException;
 import java.util.List;
+import java.util.Objects;
 
 /**
- * Converts {@link BSONObject} to {@link StructuredRecord}.
+ * Transforms {@link BSONObject} to {@link StructuredRecord}.
  */
-public class BSONConverter {
-  private static final List<Schema.Type> VALID_TYPES = Lists.newArrayList(Schema.Type.ARRAY, Schema.Type.BOOLEAN,
-                                                                          Schema.Type.BYTES, Schema.Type.STRING,
-                                                                          Schema.Type.DOUBLE,
-                                                                          Schema.Type.FLOAT, Schema.Type.INT,
-                                                                          Schema.Type.LONG, Schema.Type.NULL);
+public class BSONObjectToRecordTransformer {
+
   private final Schema schema;
 
-  public BSONConverter(Schema schema) throws IOException {
+  public BSONObjectToRecordTransformer(Schema schema) {
```

Comment:
please add unit tests for both transformers

Reply:
Added.